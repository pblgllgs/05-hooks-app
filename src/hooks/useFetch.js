import { useEffect, useRef, useState } from "react";

export const useFetch = (url) => {
    const [state, setState] = useState({
        data: null,
        loading: true,
        error: null,
    });

    const estaMontado = useRef(true);

    useEffect(() => {
        return () => {
            estaMontado.current = false;
        };
    }, []);

    useEffect(() => {
        setState({
            data: null,
            loading: true,
            error: null,
        });
        fetch(url)
            .then((resp) => resp.json())
            .then((data) => {
                setTimeout(() => {
                    if (estaMontado.current) {
                        setState({
                            loading: false,
                            error: null,
                            data,
                        });
                    }else{
                        console.log('No se llam√≥')
                    }
                }, 4000);
            });
    }, [url]);

    return state;
};
