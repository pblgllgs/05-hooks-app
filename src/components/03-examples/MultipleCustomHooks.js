import React from 'react'
import { useFetch } from '../../hooks/useFetch'
import './styles.css'

export const MultipleCustomHooks = () => {

    const state = useFetch(`https://www.breakingbadapi.com/api/quotes/1`);
    console.log(state)

    return (
        <>
            <h1>CustomHooks</h1>
        </>
    )
}