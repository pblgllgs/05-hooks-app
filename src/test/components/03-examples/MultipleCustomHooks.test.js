import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks";
import { shallow } from "enzyme";
import { useFetch } from "../../../hooks/useFetch";
import { useCounter } from "../../../hooks/useCounter";

jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("Pruebas en MultiplecustomHooks", () => {
    
    beforeEach(() => {
        useCounter.mockReturnValue({
            counter: 1,
            increment: () => { },
        });
    });

    test("Debe de mostrarse correctamente", () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null,
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de mostrar la informaciÃ³n", () => {
        useFetch.mockReturnValue({
            data: [
                {
                    author: "pablo",
                    quote: "Hola mundo",
                },
            ],
            loading: false,
            error: null,
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper.find(".alert").exists()).toBe(false);
        expect(wrapper.find(".mb-0").text().trim()).toBe("Hola mundo");
        expect(wrapper.find("figcaption").text().trim()).toBe("pablo");
    });
});
