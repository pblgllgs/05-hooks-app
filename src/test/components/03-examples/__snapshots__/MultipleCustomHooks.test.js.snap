// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MultiplecustomHooks Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    <p>
      Loading...
    </p>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</div>
`;
